components/service-operator/apis/access/v1beta1/principal_types.go

```diff
@@ -73,18 +87,32 @@ func (s *Principal) GetStackTemplate() *cloudformation.Template {
 	template.Parameters[IAMPermissionsBoundaryParameterName] = map[string]string{
 		"Type": "String",
 	}
+	template.Parameters[ServiceOperatorIAMRoleArn] = map[string]string{
+		"Type": "String",
+	}
 
 	template.Resources[IAMRoleResourceName] = &cloudformation.AWSIAMRole{
 		RoleName:                 s.GetRoleName(),
-		AssumeRolePolicyDocument: cloudformation.NewAssumeRolePolicyDocument(cloudformation.Ref(IAMRolePrincipalParameterName)),
+		AssumeRolePolicyDocument: cloudformation.NewAssumeRolePolicyDocument(cloudformation.Ref(IAMRolePrincipalParameterName), cloudformation.Ref(ServiceOperatorIAMRoleArn)),
 		PermissionsBoundary:      cloudformation.Ref(IAMPermissionsBoundaryParameterName),
 	}
 
+	template.Resources[SharedPolicyResourceName] = &cloudformation.AWSIAMPolicy{
+		PolicyName:     s.GetRoleName(),
+		PolicyDocument: cloudformation.NewRolePolicyDocument([]string{"*"}, []string{"ecr:GetAuthorizationToken"}),
```

Comment:
So any principal will be allowed to ecr:GetAuthorizationToken on any resource?

Reply:
Yes. GetAuthorizationToken exchanges your IAM creds for docker creds. If your IAM creds have no repository policies, then your docker creds will have no powers, so this is fine and as designed.